docs/intro/tutorial.rst

```diff
@@ -55,318 +55,333 @@ This will create a ``tutorial`` directory with the following contents::
-    [s]   view(response)    View response in a browser
-    
-    >>>
+In order to find the proper CSS selectors to use, you might find useful opening
+the response page from the shell in your web browser using ``view(response)``.
+You can use your browser developer tools or extensions like Firebug. For more
+information see :ref:`topics-firebug` and :ref:`topics-firefox`.
 
-After the shell loads, you will have the response fetched in a local
-``response`` variable, so if you type ``response.body`` you will see the body
-of the response, or you can type ``response.headers`` to see its headers.
 
-More importantly ``response`` has a ``selector`` attribute which is an instance of
-:class:`~scrapy.selector.Selector` class, instantiated with this particular ``response``.
-You can run queries on ``response`` by calling ``response.selector.xpath()`` or
-``response.selector.css()``. There are also some convenience shortcuts like ``response.xpath()``
-or ``response.css()`` which map directly to ``response.selector.xpath()`` and
-``response.selector.css()``.
+XPath: a brief intro
+^^^^^^^^^^^^^^^^^^^^
 
+Besides `CSS`_, Scrapy selectors also support using `XPath`_ expressions::
 
-So let's try it::
+    >>> response.xpath('//title')
+    [<Selector xpath='//title' data=u'<title>Quotes to Scrape</title>'>]
+    >>> response.xpath('//title/text()').extract_first()
+    u'Quotes to Scrape'
 
-    In [1]: response.xpath('//title')
-    Out[1]: [<Selector xpath='//title' data=u'<title>Quotes to Scrape</title>'>] 
-    
-    In [2]: response.xpath('//title').extract()
-    Out[2]: [u'<title>Quotes to Scrape</title>']
-    
-    In [3]: response.xpath('//title/text()')
-    Out[3]: [<Selector xpath='//title/text()' data=u'Quotes to Scrape'>]
+XPath expressions are very powerful, and are the foundation of Scrapy
+Selectors. In fact, CSS selectors are converted to XPath under-the-hood. You
+can see that if you read closely the text representation of the selector
+objects in the shell.
 
-    In [4]: response.xpath('//title/text()').extract()
-    Out[4]: [u'Quotes to Scrape']
-    
-    In [11]: response.xpath('//title/text()').re('(\w+)')
-    Out[11]: [u'Quotes', u'to', u'Scrape']
+While perhaps not as popular as CSS selectors, XPath expressions offer more
+power because besides navigating the structure, it can also look at the
+content. Using XPath, you're able to select things like: **select the link
+that contains the text "Next Page"**. This makes XPath very fitting to the task
+of scraping, and we encourage you to learn XPath even if you already know how to
+construct CSS selectors, it will make scraping much easier.
 
-Extracting the data
-^^^^^^^^^^^^^^^^^^^
+We won't cover much of XPath here. To learn more about XPath, we recommend
+`this tutorial to learn XPath through examples
+<http://zvon.org/comp/r/tut-XPath_1.html>`_, and `this tutorial to learn "how
+to think in XPath" <http://plasmasturm.org/log/xpath101/>`_.
```

Comment:
Scrapy docs also have a section about selectors which is mostly focused on xpaths, what do you think about adding a link?